app/src/main/java/com/bashkevich/androidfundamentals/moviesdetails/view/FragmentMoviesDetails.kt

```diff
@@ -38,40 +47,49 @@ class FragmentMoviesDetails : Fragment() {
         val ageCategoryView = view.findViewById<TextView>(R.id.age_category)
         val titleView = view.findViewById<TextView>(R.id.title_details)
         val genresView = view.findViewById<TextView>(R.id.genres_details)
-        val ratingView = view.findViewById<AppCompatRatingBar>(R.id.movie_rating_details)
+        val ratingView = view.findViewById<RatingBar>(R.id.movie_rating_details)
         val reviewView = view.findViewById<TextView>(R.id.reviews_details)
         val descriptionView = view.findViewById<TextView>(R.id.description_details)
         val castView = view.findViewById<TextView>(R.id.cast)
         actorsRecyclerView = view.findViewById(R.id.actors_recycler_view)
 
+        movieId?.let { moviesDetailsViewModel.getMovieFromList(it) }
 
-        movie?.let { movie ->
-            Picasso.get().load(Uri.parse(movie.backdrop)).into(backdropView)
 
-            ageCategoryView.text = context?.getString(R.string.age_category, movie?.minimumAge)
-            titleView.text = movie.title
-            genresView.text = movie.genres.joinToString { it.name }
-            ratingView.rating = movie.ratings.div(2)!!
-            reviewView.text = context?.getString(R.string.reviews, movie?.numberOfRatings)
-            descriptionView.text = movie.overview
+        moviesDetailsViewModel.movieLiveData.observe(this.viewLifecycleOwner) {movie->
 
-            val actors = movie.actors
+            movie?.let { movie ->
+                Picasso.get().load(Uri.parse(movie.backdrop)).into(backdropView)
 
-            if (actors.isEmpty()) {
-                castView.visibility = View.GONE
-            }
+                ageCategoryView.text = context?.getString(R.string.age_category, movie.minimumAge)
+                titleView.text = movie.title
+                genresView.text = movie.genres.joinToString { it.name }
+                ratingView.rating = movie.ratings / 2
+                reviewView.text = context?.getString(R.string.reviews, movie.numberOfRatings)
+                descriptionView.text = movie.overview
+
+                setUpActorsRecyclerView()
```

Comment:
Do you want to setup recycler every time when movie is selected?

Reply:
I load it when actors  are not empty